video/out/drm_common.c

```diff
@@ -715,21 +793,52 @@ static void kms_show_connector_modes_callback(struct mp_log *log, int card_no,
     mp_info(log, "\n");
 }
 
-static void kms_show_available_connectors_and_modes(struct mp_log *log, int card_no)
+static void kms_show_available_connectors_and_modes(struct mp_log *log,
+                                                    int card_no,
+                                                    const char *card_path)
 {
-    kms_show_foreach_connector(log, card_no, kms_show_connector_modes_callback);
+    kms_show_foreach_connector(log, card_no, card_path,
+                               kms_show_connector_modes_callback);
 }
 
 static void kms_show_foreach_card(
-    struct mp_log *log, void (*show_fn)(struct mp_log*,int))
+    struct mp_log *log, void (*show_fn)(struct mp_log*,int,const char *))
 {
-    for (int card_no = 0; card_no < DRM_MAX_MINOR; card_no++) {
-        int fd = open_card(card_no);
-        if (fd < 0)
-            break;
+    drmDevice *devices[DRM_MAX_MINOR] = { 0 };
+    int card_count = drmGetDevices(devices,
+                                   MP_ARRAY_SIZE(devices));
+    if (card_count < 0) {
+        mp_err(log, "Listing DRM devices with drmGetDevices failed! (%s)\n",
+               mp_strerror(errno));
+    }
+
+    for (int card_no = 0; card_no < card_count; card_no++) {
+        drmDevice *dev = devices[card_no];
+
+        if (!dev || !(dev->available_nodes & (1 << DRM_NODE_PRIMARY)))
+            continue;
+
+        char *primary_node_path = dev->nodes[DRM_NODE_PRIMARY];
+        if (!primary_node_path || !primary_node_path[0]) {
```

Comment:
Same comments apply here - please kill off the `!dev` and `!primary_node_path || !primary_node_path[0]` checks.